### ISSUE TITLE
check_ping -  CRITICAL - Could not interpret output from ping command on Cent 8+ systems

### ISSUE THREAD

**Issue by sawolf** [nagios-plugins/nagios-plugins#647]
See https://github.com/NagiosEnterprises/nagioscore/issues/855 - I've seen this issue as well while working on unrelated problems. Posting the bug here until I have a chance to look into it.

**Comment by chicks-net** [comment]
It would be better if this issue included a complete description of your problem.  Do you know which version of `ping` got installed in the update?

**Comment by gnw3** [comment]
Using Fedora 36 I see: 
```
SERVICE ALERT: localhost;PING;UNKNOWN;HARD;1;CRITICAL - Could not interpret output from ping command
````

```
 % ping -V
ping from iputils 20211215
 % ping -c 1 localhost
PING localhost(localhost (::1)) 56 data bytes
64 bytes from localhost (::1): icmp_seq=1 ttl=64 time=0.034 ms

--- localhost ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.034/0.034/0.034/0.000 ms
```

**Comment by IOMadness** [comment]
Sorry for posting on a older thread, however I see no development and I've encountered this same issue w/ nagios-plugins 2.4.12 on a RHEL9 system. Doesn't seem to occur on older (7/8) RHEL.

Basically, on my system, when running the check_ping plugin against either the FQDN of the local system (pointing to an IPv4 address) or "localhost", I get the following:

CRITICAL - Could not interpret output from ping command


Turning up the verbosity, I can see the following:
CMD: /usr/sbin/ping6 -n -U -w 35 -c 5 myhost.com
Got stderr: /usr/sbin/ping6: my.host.com: Address family for hostname not supported


However, running it against an IPv4 IP works just fine:

[root@myhost nagios-plugins-2.4.12]# plugins/check_ping -H 10.11.12.13 -w 5000,80% -c 6000,90% -vvvv
CMD: /usr/sbin/ping -n -U -w 35 -c 5 10.11.12.13
Output: PING 10.11.12.13 (10.11.12.13) 56(84) bytes of data.
Output: 64 bytes from 10.11.12.13: icmp_seq=1 ttl=64 time=0.040 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=2 ttl=64 time=0.061 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=3 ttl=64 time=0.062 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=4 ttl=64 time=0.061 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=5 ttl=64 time=0.051 ms
Output: 
Output: --- 10.11.12.13 ping statistics ---
Output: 5 packets transmitted, 5 received, 0% packet loss, time 4132ms
Output: rtt min/avg/max/mdev = 0.040/0.055/0.062/0.008 ms
PING OK - Packet loss = 0%, RTA = 0.05 ms|rta=0.055000ms;5000.000000;6000.000000;0.000000 pl=0%;80;90;0
5000.000000:80% 6000.000000:90%


You can see when using an IPv4 IP, it uses /usr/sbin/ping instead of /usr/sbin/ping6. This error only seems to happen when referencing the local host's FQDN or "localhost", I do not have issues running it against remote hostnames w/ IPv4 addresses.

WORKAROUND:
I disabled IPv6 addressing in the kernel:

Create a new file and add parameters:
vim /etc/sysctl.d/ipv6.conf
# Disable IPv6 for all interfaces
net.ipv6.conf.all.disable_ipv6 = 1

# Disable IPv6 for each network interface
net.ipv6.conf.lo.disable_ipv6 = 1
net.ipv6.conf.ens1.disable_ipv6 = 1

...next, backup and rebuild initramfs
cp -p /boot/initramfs-$(uname -r).img /boot/initramfs-$(uname -r).old.img
dracut -f -v

reboot the system.


Now, running the command against the local FQDN uses /usr/bin/ping appropriately:
[root@arcc-nagios-dev nagios-plugins-2.4.12]# plugins/check_ping -H myhost.com -w 5000,80% -c 6000,90% -vvvv
CMD: /usr/sbin/ping -n -U -w 35 -c 5 myhost.com
Output: PING myhost.com (10.11.12.13) 56(84) bytes of data.
Output: 64 bytes from 10.11.12.13: icmp_seq=1 ttl=64 time=0.036 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=2 ttl=64 time=0.080 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=3 ttl=64 time=0.080 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=4 ttl=64 time=0.076 ms
Output: 64 bytes from 10.11.12.13: icmp_seq=5 ttl=64 time=0.078 ms
Output: 
Output: --- myhost.com ping statistics ---
Output: 5 packets transmitted, 5 received, 0% packet loss, time 4122ms
Output: rtt min/avg/max/mdev = 0.036/0.070/0.080/0.017 ms
PING OK - Packet loss = 0%, RTA = 0.07 ms|rta=0.070000ms;5000.000000;6000.000000;0.000000 pl=0%;80;90;0
5000.000000:80% 6000.000000:90%
